File: solutions/php/flower-field/1/FlowerField.php
<?php
declare(strict_types=1);

class FlowerField
{
    private array $garden;
    private int $rows;
    private int $cols;

    public function __construct(array $garden)
    {
        $this->garden = $garden;
        $this->rows = count($garden);
        $this->cols = $this->rows > 0 ? strlen($garden[0]) : 0;
    }

    public function annotate(): array
    {
        $result = [];
        
        for ($r = 0; $r < $this->rows; $r++) {
            $row = '';
            for ($c = 0; $c < $this->cols; $c++) {
                if ($this->garden[$r][$c] === '*') {
                    $row .= '*';
                } else {
                    $count = $this->countAdjacentFlowers($r, $c);
                    $row .= $count > 0 ? (string)$count : ' ';
                }
            }
            $result[] = $row;
        }
        
        return $result;
    }

    private function countAdjacentFlowers(int $r, int $c): int
    {
        $count = 0;
        $directions = [
            [-1,-1],[-1,0],[-1,1],
            [0,-1],        [0,1],
            [1,-1], [1,0], [1,1]
        ];
        
        foreach ($directions as [$dr, $dc]) {
            $nr = $r + $dr;
            $nc = $c + $dc;
            if ($nr >= 0 && $nr < $this->rows && $nc >= 0 && $nc < $this->cols) {
                if ($this->garden[$nr][$nc] === '*') {
                    $count++;
                }
            }
        }
        return $count;
    }
}
?>

File: solutions/ruby/flower-field/1/flower_field.rb
class FlowerField
  # Cambiado de self.transform a self.annotate para que coincida con el test
  def self.annotate(board)
    # 1. Validaciones básicas de estructura
    return board if board.empty? || board.first.empty?
    raise ArgumentError unless valid_board?(board)

    rows = board.length
    cols = board.first.length

    # Convertimos las cadenas en una matriz de caracteres
    matrix = board.map(&:chars)

    # 2. Recorremos cada celda de la matriz
    (0...rows).map do |r|
      (0...cols).map do |c|
        if matrix[r][c] == '*'
          '*' 
        else
          count = count_adjacent_flowers(matrix, r, c, rows, cols)
          count > 0 ? count.to_s : ' ' 
        end
      end.join 
    end
  end

  private

  def self.valid_board?(board)
    board.map(&:length).uniq.size == 1
  end

  def self.count_adjacent_flowers(matrix, r, c, max_rows, max_cols)
    flowers = 0

    (-1..1).each do |dr|
      (-1..1).each do |dc|
        next if dr == 0 && dc == 0 

        neighbor_r = r + dr
        neighbor_c = c + dc

        if neighbor_r.between?(0, max_rows - 1) && neighbor_c.between?(0, max_cols - 1)
          flowers += 1 if matrix[neighbor_r][neighbor_c] == '*'
        end
      end
    end

    flowers
  end
end